packages/dcos-launch/extra/launch/onprem.py

```diff
@@ -0,0 +1,138 @@
+import copy
+import logging
+import subprocess
+
+import retrying
+import yaml
+
+import launch.aws
+import launch.util
+
+import test_util.aws
+import test_util.onprem
+import test_util.runner
+from test_util.helpers import Url
+
+log = logging.getLogger(__name__)
+
+STATE_FILE = 'LAST_COMPLETED_STAGE'
+
+
+@retrying.retry(wait_fixed=1000, stop_max_delay=120000)
+def check_ssh(ssher, host, port):
+    ssher.get_home_dir(host, port)
+
+
+class OnpremLauncher(launch.util.AbstractLauncher):
```

Comment:
Funny that you ask this; I agree and already merged a PR to make all Launchers consume the config and each of the functions use self.config: https://github.com/dcos/dcos/pull/1485/commits/2c75b6868691cde37bc01c4d7f233789633640e9